OPHD/UI/PopulationPanel.h

```diff
@@ -55,4 +55,6 @@ class PopulationPanel : public Control
 	int mResidentialCapacity{0};
 	int mCrimeRate{0};
 	int mPopulationPanelWidth{0};
+
+	void init();
```

Comment:
I generally recommend against `init` methods, as what they contain should really be in a constructor.

These did make sense in the past, when you wanted to share code between constructors, which I see is the direction you were going here. Since C++11 this approach is no longer necessary, since "Delegating Constructors" were introduced to the language. When you want to re-use constructors, you can call one from the other using the member initialization list. In that case, only that one single "member" can be initialized.

Some details are found about half way down this page:
https://en.cppreference.com/w/cpp/language/constructor

One approach might be for the default constructor to call a constructor taking parameters, and passing default values. However, in that case, it might make sense to set default values for the parameters, making them optional, and allowing that constructor to also function as the default constructor.

In this particular case, that's possible if you're still using pointer syntax, but not reference syntax. You can't quite switch to reference syntax until after the setters are removed, since a reference is basically a `const` pointer (to data that may or may not be `const`).

Reply:
Generally agreed here -- Part of the style that I've been using lately is "Only 5 Lines of Code" per function (including c'tor) and I tend to extract some of that code into stand alone functions that are wrapped in an anonymous namespace.

Reply:
My general rule of thumb is to try and keep functions within one page of code, so you never need to scroll to see the full context. There's a lot less you need to remember when you can see it all at once.